src/main/java/redis/clients/jedis/HostAndPort.java

```diff
@@ -47,6 +47,38 @@ public String toString() {
     return host + ":" + port;
   }
 
+  /**
+   * Parses String.
+   * String must be in ( host + ":" + port ) format.
+   * Port is optional
+   * @param from String to parse
+   * @return array of host and port strings
+     */
+  public static String[] extractParts(String from){
+    int idx     = from.lastIndexOf(":");
+    String host = idx != -1 ? from.substring(0, idx)  : from;
+    String port = idx != -1 ? from.substring(idx + 1) : "";
+    return new String[] { host, port };
+  }
+
+  /**
+   * Creates HostAndPort instance from string.
+   * String must be in ( host + ":" + port ) format.
+   * Port is mandatory.
+   * @param from String to parse
+   * @return HostAndPort instance
+     */
+  public static HostAndPort parseString(String from){
```

Comment:
agree, but there is redis's IPv6-related bug (or maybe feature) https://github.com/antirez/redis/issues/3076. That's why I see no possibility to split IPv6 address in 'correct' way. I will add Comment + TODO.